vllm/model_executor/model_loader/loader.py

```diff
@@ -422,6 +422,16 @@ def get_end_ptr(tensor: torch.Tensor) -> int:
                     result[k] = t
         return result
 
+    def _prepare_weights(self, model_name_or_path: str,
+                         revision: Optional[str]):
+        if os.path.isdir(model_name_or_path):
+            return model_name_or_path
+        else:
+            allow_patterns = ["*.safetensors"]
+            return download_weights_from_hf(model_name_or_path,
+                                            self.load_config.download_dir,
+                                            allow_patterns, revision)
```

Comment:
Do we always assume the model repo has weights in `.safetensors` format?

Reply:
currently yes, because this loader is only compatible with the saved sharded states, which is done using safetensors.

Reply:
I see - should we be explicit about it if users specify sharded states as the load format? (Not sure if I missed it if such a warning/info is already there) 

What happens if the model repo does not have `.safetensors`?

Reply:
we currently have a check at https://github.com/vllm-project/vllm/blob/main/vllm/model_executor/model_loader/loader.py#L451